### ISSUE TITLE
Running ssh with a command (without attaching) against a Windows host causes glitches in all Windows consoles

### ISSUE THREAD

**Issue by doctorpangloss** [rmyorston/busybox-w32#288]
# Summary

In certain situations, ASCII control characters are emitted onto the shell prompt; unusual characters and misprinting appears; or other console glitches depending on the console and configuration used with `busybox64.exe bash`.

These can be categorized into two issues that can be found elsewhere on the Internet:
 - The **`[A[B[C[D`** issue: Hitting the arrow keys on the text prompt emits ANSI control characters instead of performing a typical arrow key keyboard input.
 - The **Music Note** issue: The shell emits music note character, dot characters and other ASCII representations of control code, and doesn't correctly clear output when performing arrow key, return and other similar keyboard inputs.

This ticket will focus on interactions between all consoles, `ssh` and `busybox64.exe bash`.

# Description
Executing commands with `ssh` on a Windows host configured to use `busybox64.exe bash` as its default SSH shell glitches the host shell and makes it no longer usable.

### Expected (in `cmd` shell)

Click thumbnail to start gif
![cmd_Dy6YjiKmvX](https://user-images.githubusercontent.com/2229300/219463471-b87057d7-279d-4b56-95c9-73fc6d79d1d8.gif)

### Observed (in `busybox64.exe bash -il`)

Click thumbnail to start gif
![cmd_p4Frs7sK2N](https://user-images.githubusercontent.com/2229300/219463731-a6f27c39-d8e5-427f-8f07-eb6a49117ca7.gif)

# Reproduction

1. Provision an OpenSSH accessible Windows host with public key authentication. The host can be the same machine as the client. Install the OpenSSH Server Windows optional feature.
     
![install-openssh-server-windows10-settings](https://user-images.githubusercontent.com/2229300/219464868-38746879-cb07-40f7-b2f8-e4495c8e4502.jpg)

2. Install `busybox64.exe` on this Windows host.
3. Provision SSH keys. For your convenience, this script will provision your GitHub keys (run as administrator inside busybox64 bash):

```sh
mkdir -pv ~/.ssh
mkdir -pv C:/ProgramData/ssh
curl https://github.com/rmyorston.keys | tee ~/.ssh/authorized_keys C:/ProgramData/ssh/administrators_authorized_key
icacls "C:/ProgramData/ssh/administrators_authorized_keys" /inheritance:r /grant "Administrators:F" /grant "SYSTEM:F"
```
4. Start the `sshd` service (run as administrator):
```sh
net start sshd
```
5. On the client, install the OpenSSH Client Windows optional feature, add your SSH keys to GitHub, start `busybox64.exe bash -il`, then execute the following adjusting your username and destination busybox location:
```sh
ssh administrator@target-host -- C:/busybox64.exe sh -c 'echo Observe bad behavior'
```
6. Observe your terminal is now glitched. It shouldn't be.
7. Observe the issue does not reproduce when running step 5 inside `cmd`.

**Comment by doctorpangloss** [comment]
Some additional notes:

 - this reproduces with PowerShell on the remote host
 - you can try this without the public key provisioning step, but it may prompt for a password and input possibly prevents reproduction
 - this might reproduce against Linux SSH hosts too

Based on your earlier investigation, it's probably an issue with SSH on Windows.

**Comment by rmyorston** [comment]
Thanks for the detailed instructions!

As with many things on Microsoft Windows, I've no idea what's going on here. However, I did observe that after running a remote command with ssh the console was left in a strange mode will all its flags cleared to zero.

I've added some code to the busybox-w32 shell to check for this condition and attempt to recover the situation. This doesn't seem to work in legacy consoles, but I think those aren't the default these days.

Try the latest prelease binary: [busybox_pre64.exe](https://frippery.org/files/busybox/prerelease/busybox_pre64.exe).
